pkg/controllers/subnet/subnet_controller.go

```diff
@@ -54,15 +54,26 @@ func (r *SubnetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	defer func() {
 		log.Info("Finished reconciling Subnet", "Subnet", req.NamespacedName, "duration(ms)", time.Since(startTime).Milliseconds())
 	}()
-	metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerSyncTotal, MetricResTypeSubnet)
 
+	statusUpdater := common.StatusUpdater{
+		Client:          r.Client,
+		NSXConfig:       r.SubnetService.NSXConfig,
+		Recorder:        r.Recorder,
+		MetricResType:   MetricResTypeSubnet,
+		NSXResourceType: "Subnet",
+		ResourceType:    "Subnet",
+	}
+
+	metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerSyncTotal, MetricResTypeSubnet)
```

Comment:
What do you feel if we also move the call of `metrics.CounterInc` to StatusUpdater ? If the effort is huge, we can update it in future with another change.

Reply:
It's fine to include in this PR, updated, thanks!